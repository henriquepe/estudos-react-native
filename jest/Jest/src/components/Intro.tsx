import React, {useState} from 'react';
import {Text, View, TextInput} from 'react-native';

export const Intro: React.FC = () => {
  const [name, setName] = useState('');

  return (
    <View>
      <Text testID="hello">Hello {name}</Text>

      <View>
        <TextInput
          testID="input"
          placeholder="Digite seu nome"
          onChangeText={setName}
        />
      </View>

      <Text testID="name">O nome que vocÃª digitou foi: {name}</Text>
    </View>
  );
};
