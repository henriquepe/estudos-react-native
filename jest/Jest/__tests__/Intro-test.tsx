// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import {Intro} from '../src/components/Intro';
import {render, fireEvent} from '@testing-library/react-native';

it('render correctly', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('render the text Hello with the name of the user', () => {
  const {getByTestId} = render(<Intro />);

  const inputComponent = getByTestId('input');

  const testText = 'Henrique';

  fireEvent.changeText(inputComponent, testText);

  const textComponent = getByTestId('hello');

  expect(textComponent.props.children).toContain('Henrique');
});

it('render the username right side of the message', () => {
  const {getByTestId} = render(<Intro />);

  const inputComponent = getByTestId('input');

  const testText = 'Henrique';

  fireEvent.changeText(inputComponent, testText);

  const textComponent = getByTestId('name');

  expect(textComponent.props.children).toContain('Henrique');
});
