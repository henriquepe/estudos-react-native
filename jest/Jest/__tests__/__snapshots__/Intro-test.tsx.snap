// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View>
  <Text
    testID="hello"
  >
    Hello 
    
  </Text>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Digite seu nome"
      rejectResponderTermination={true}
      testID="input"
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    testID="name"
  >
    O nome que vocÃª digitou foi: 
    
  </Text>
</View>
`;
